### DISCUSSION TITLE
Angular/TS support

### DISCUSSION THREAD

**Discussion by Malakhib** [sandworm-hq/sandworm-js#65] [Ideas]
Is there any way we could get support for Angular/TypeScript?

This is a very beneficial tool and I'm sure there's many like me using Angular that would appreciate the ability to use it with their projects.

Thanks!

**Comment by andreimarinescu** [comment]
Hey, @Malakhib! Thanks for reaching out! Definitely - we do want to provide as much support as possible for Angular and Typescript projects in general.

While we haven't tested this specific scenario yet, I can't think off the top of my head of anything preventing Guard to work with any Angular or TS project.

We'll do some tests on our side as well, but I'd be curious to know what your experience has been if you do end up giving it a try in your project. I'm sure we could for example automatically trust some modules, like we would for react-dom for example (https://docs.sandworm.dev/guard#trusted-modules), but we'd be more than happy to hear any other ideas you might have.

**Reply by Malakhib** [reply]
My apologies, you are correct.
Upon further inspection, I was misdiagnosing a few issues as incompatibility; however, I can confirm it works now.
Thank you!

**Reply by andreimarinescu** [reply]
That's really great to hear, @Malakhib! If you do end up hitting any roadblocks please reach out, we're more than happy to help!

**Reply by Malakhib** [reply]
Hello,

My apologies for it taking a while to get back to this, I got distracted with a few other projects.

After digging into things a bit deeper, I believe I was able to get sandworm-audit and sandworm-jest working however I seem to still be having troubles with sandworm-guard-js.
After fiddling with polyfills [1] and some dist/index.d.ts , and TS, compatibility issues [2] the closest I got was it launching but setting off 404s and 413s [3].

I'm currently running Angular 14.2.12, Node 16.10.0, and npm 7.24.0, and typescript 4.7.4.

Thank you!

[1]
![TS_Issues](https://user-images.githubusercontent.com/71473952/229203441-19b1eb87-e531-47fc-bc09-c0eff68fdfaf.PNG)


[2]
![polyfill](https://user-images.githubusercontent.com/71473952/229203386-ee86391d-25ce-42e8-8cf5-b356aa713ef6.PNG)


[3]
![404_413](https://user-images.githubusercontent.com/71473952/229186449-6d6414fe-ff7a-4f7b-bc2a-562e3c1c3586.PNG)

**Reply by andreimarinescu** [reply]
Hey, @Malakhib . Thanks for setting this test up. We'll try to get this running on our setup as well and sort out the kinks / offer some guidance on how to set up Guard efficiently using Angular and TS. Might take a couple of days to get back on this but I'll post an update as soon as we have one.
